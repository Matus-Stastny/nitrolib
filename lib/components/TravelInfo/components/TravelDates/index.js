"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _Text = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Text"));

var _Calendar = _interopRequireDefault(require("@kiwicom/orbit-components/lib/icons/Calendar"));

var _Stack = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Stack"));

var _Heading = _interopRequireDefault(require("@kiwicom/orbit-components/lib/Heading"));

var _index = _interopRequireDefault(require("../../../Translate/index"));

var _ref2 =
/*#__PURE__*/
React.createElement(_Heading.default, {
  element: "h4",
  type: "title3",
  spaceAfter: "medium"
}, React.createElement(_index.default, {
  t: "holidays.detail.travel_dates"
}));

var _ref3 =
/*#__PURE__*/
React.createElement(_Calendar.default, {
  color: "primary"
});

var _ref4 =
/*#__PURE__*/
React.createElement("br", null);

var TravelDates = function TravelDates(_ref) {
  var data = _ref.data;
  var from = data.from,
      to = data.to;
  return React.createElement(React.Fragment, null, _ref2, React.createElement(_Stack.default, {
    direction: "row",
    spacing: "compact"
  }, _ref3, React.createElement(_Text.default, null, React.createElement(_index.default, {
    t: "holidays.hotel_tags.from_x_till_x_",
    values: {
      from: from,
      to: to
    },
    html: true
  }), _ref4, React.createElement(_index.default, {
    t: "holidays.detail.travel_dates.x_nights_in_resort",
    values: {
      num: 11
    }
  }))));
};

var _default = TravelDates;
exports.default = _default;