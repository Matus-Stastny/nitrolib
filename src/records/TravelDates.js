// @flow strict

export type TravelDates = {|
  from: string,
  to: string,
|};
