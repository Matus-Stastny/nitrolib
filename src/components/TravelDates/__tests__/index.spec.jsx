// @flow
import * as React from "react";
import { shallow } from "enzyme";

import TravelDates from "..";

describe("#TravelDates", () => {
  test("render", () => {
    const data = {
      from: "Fri 20 Nov",
      to: "Sun 4 Dec",
    };
    const wrapper = shallow(<TravelDates data={data} />);

    expect(wrapper).toMatchSnapshot();
  });
});
