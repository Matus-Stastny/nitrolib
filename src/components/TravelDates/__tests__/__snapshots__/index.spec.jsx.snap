// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#TravelDates render 1`] = `
<Fragment>
  <Heading
    element="h4"
    spaceAfter="medium"
    type="title3"
  >
    <Translate
      html={false}
      t="holidays.detail.travel_dates"
      transform={[Function]}
      values={Object {}}
    />
  </Heading>
  <Stack
    direction="row"
    spacing="compact"
  >
    <Calendar
      color="primary"
    />
    <Text>
      <Translate
        html={true}
        t="holidays.from_x_till_x_"
        transform={[Function]}
        values={
          Object {
            "from": "Fri 20 Nov",
            "to": "Sun 4 Dec",
          }
        }
      />
      <br />
      <Translate
        html={false}
        t="holidays.x_nights_in_restort"
        transform={[Function]}
        values={
          Object {
            "num": 11,
          }
        }
      />
    </Text>
  </Stack>
</Fragment>
`;
